modules/ui/cmd.js

```diff
@@ -14,14 +14,14 @@ export var uiCmd = function (code) {
     if (detected.os === 'win') {
         if (code === '⌘⇧Z') return 'Ctrl+Y';
     }
-
+    var mac = (detected.os === 'mac');
     var result = '',
         replacements = {
-            '⌘': 'Ctrl',
-            '⇧': 'Shift',
-            '⌥': 'Alt',
-            '⌫': 'Backspace',
-            '⌦': 'Delete'
+            '⌘': mac ? '⌘ '  : t('shortcuts.key.ctrl'),
+            '⇧': mac ? '⇧ '  : t('shortcuts.key.shift'),
+            '⌥': mac ? '⌥ '  : t('shortcuts.key.alt'),
+            '⌫': mac ? '⌫ '  : t('shortcuts.key.backspace'),
+            '⌦': mac ? '⌦ '  : t('shortcuts.key.del'),
```

Comment:
This is essentially the same as what c22fa60b8a33fdfd9dc668619a797f8c2889f9ad did in `uiCmd.display` down below to fix the Keyboard Shortcuts screen. I suspect c22fa60b8a33fdfd9dc668619a797f8c2889f9ad didn’t touch this function because it’s also being used in places where raw key identifiers are needed rather than human-readable labels. For example, the following code uses the return value to register the actual keyboard shortcut:

https://github.com/openstreetmap/iD/blob/3025d4f4217238ecd57dbb09734c9d9e93124569/modules/ui/full_screen.js#L72-L73 https://github.com/openstreetmap/iD/blob/3025d4f4217238ecd57dbb09734c9d9e93124569/modules/ui/zoom.js#L102-L110

If you run this branch on macOS, or on Windows with the German localization, will the shortcuts function correctly when you press them? If not, then maybe we need to keep this function as is. If we look back at the code I pointed to earlier, notice how it’s only being used to create the tooltip – so it’s probably only for display purposes:

https://github.com/openstreetmap/iD/blob/b5d45e3f28ccb9a39d243e7504415152e1ce8ec2/modules/ui/sections/data_layers.js#L404

If that’s the case, then changing that code to call `uiCmd.display` instead might fix the issue for that particular tooltip. So instead of implementing anything new, you’d be reviewing each of the calls to `uiCmd`, deciding whether it should be replaced with a call to `uiCmd.display`.

Reply:
Thank very much for the detailed explanation, it's very helpful and I now understood with more clarity, gotta read more about this codebase, and I've marked the calls where uiCmd been called and now checking how useful, the uiCmd.display would be...will submit a pull request again after high detailed review of each call.

Reply:
1. Issue with Key Functionality.

It seems that the keyboard shortcut for ? isn’t functioning as expected. Based on testing, it appears the shortcut should be labeled as Shift + ? instead of just ?, as pressing ? alone triggers the black overlay below the inspect button in the top-left corner.

<img width="1216" alt="image" src="https://github.com/user-attachments/assets/cf4cddbe-16f8-4558-acaf-f987484c61fa">

---

2. Interaction Limitation with Quality Assurance Tab
When the Quality Assurance tab is open in the Help section, the Keyboard Shortcuts option becomes unclickable. Interestingly, this behavior is not observed with other tabs, which allow seamless access to the keyboard shortcuts.

Image 1 : For example, the Keyboard Shortcuts options is clickable in case of points tab.
<img width="1216" alt="image" src="https://github.com/user-attachments/assets/2a98b414-7816-4d50-8169-3c2c0950e19d">


Image 2 : When the Quality Assurance tab is open in the Help section, the Keyboard Shortcuts option becomes unclickable. 
![WhatsApp Image 2024-12-03 at 11 03 13](https://github.com/user-attachments/assets/8aacaa74-b384-4dbf-87ce-f6c73c49bae0)

Reply:
<img width="1377" alt="image" src="https://github.com/user-attachments/assets/f30f20c2-69ef-4579-86e1-f31a9a3d591e">

I fixed that labelling issue mentioned as issue 1 on previous, should I create this as a entirely different issue and raise a PR or it's okay to do it here?

Reply:
> It seems that the keyboard shortcut for ? isn’t functioning as expected. Based on testing, it appears the shortcut should be labeled as Shift + ? instead of just ?, as pressing ? alone triggers the black overlay below the inspect button in the top-left corner.

I think spelling it as just “?” is a standard convention, at least on macOS. If you’re using an English keyboard layout, pressing that key alone would be <kbd>/</kbd>, not <kbd>?</kbd>. A user-selectable keyboard layout could conceivably have <kbd>?</kbd> as the primary function of a key, or it could require a modifier key other than Shift. All that matters to a Web application like iD is whether the user has pressed the some key combination that would input “?” into a text field.

Reply:
> When the Quality Assurance tab is open in the Help section, the Keyboard Shortcuts option becomes unclickable.

I can reproduce this issue – you can only click on a tiny sliver at the bottom of the button. This is probably unrelated to displaying localized keyboard shortcuts in the tooltip, so please open a separate issue about it. Thanks!

Reply:
Thanks for reviewing! I’ll go ahead and create a separate issue for this fix and then focus on the `uiCmd` functionality.

Reply:
> > It seems that the keyboard shortcut for ? isn’t functioning as expected. Based on testing, it appears the shortcut should be labeled as Shift + ? instead of just ?, as pressing ? alone triggers the black overlay below the inspect button in the top-left corner.
> 
> I think spelling it as just “?” is a standard convention, at least on macOS. If you’re using an English keyboard layout, pressing that key alone would be /, not ?. A user-selectable keyboard layout could conceivably have ? as the primary function of a key, or it could require a modifier key other than Shift. All that matters to a Web application like iD is whether the user has pressed the some key combination that would input “?” into a text field.

You're absolutely right—using "?" as the shortcut makes sense, especially since it's about the input rather than the exact key combination. It’s a good reminder that different keyboard layouts might handle this differently, Thanks for pointing that out. 😊

Reply:
<img width="704" alt="image" src="https://github.com/user-attachments/assets/7be1dd3c-7071-462e-8977-12da00f4d8f4">

if we check console.log(uiCmd('⌘Z'));  // Outputs either '⌘Z' on Mac or 'Ctrl+Z' on Windows/Linux

and now we replace that with console.log(uiCmd.display('⌘Z')); // Outputs either '⌘Z' on Mac or '⌘Z' on Windows/Linux

Also it outputs 'CTRL+Z' on German localization or any other regional language, like in case of german it should be STRG+Z when using uiCmd.display but this function shows Outputs either '⌘Z' on Mac or '⌘Z' on Windows/Linux.

I tried to fix this bug using the localization function t(), and it worked with uiCmd but not with uiCmd.display. I'm a bit unsure whether using uiCmd.display for this undo action is a bad idea, or if it's an exception due to hardcoding, or if there's something else I'm missing.